Ba persistence refactor

Enquier opened the pull request.

HuiJun commented:
Needs test

HuiJun commented on mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/webscripts/ModelsGet.java:
```diff
@@ -219,5 +215,35 @@ private JSONObject handleRequest(WebScriptRequest req, final Long maxDepth)
+            uniqueElements.add(elementsToFindJson.get(i));
+        }
+        Long maxDepth = 100000L;
+        EmsNodeUtil.handleMountSearch(mountsJson, false, false, maxDepth, uniqueElements, found, null, null);
```
handleMountSearch calls postgres for children for every single element.

BA-MBSE committed:
Decouples Federated Persistence layer from controller and service layers.
Introduces a unified persistence layer interface with a federated implementation.
Added msosa, view, and oauth modules

BA-MBSE committed:
Remove redundant commit save

BA-MBSE committed:
CodeQL updates

BA-MBSE committed:
More CodeQL changes

BA-MBSE committed:
Added timeout to database delete statement

BA-MBSE committed:
Removing overarching transactional annotations

BA-MBSE committed:
Removed transactional annotations from DefaultPermissionService

BA-MBSE committed:
Removed @Transactional annotation from modules that are not tasked with persistence implementation

BA-MBSE committed:
Fixes for failing crud tests

BA-MBSE committed:
Increased request delay for crud tests

BA-MBSE committed:
Working on fixing permissions test failures

BA-MBSE committed:
increased delay in permissions test

BA-MBSE committed:
Fixing problem with bean creation for permissions delegates that was preventing them from getting transaction proxies created

BA-MBSE committed:
Fixed artifacts behavior

BA-MBSE committed:
extending artifacts test delay to 1000 ms

BA-MBSE committed:
Fixing elastic test failures

BA-MBSE committed:
Fixing groups test failures

BA-MBSE committed:
Merge branch 'develop' into ba_persistence_refactor
